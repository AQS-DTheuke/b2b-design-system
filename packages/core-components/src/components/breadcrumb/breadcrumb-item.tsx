import { Component, Prop, Host, h, Event, EventEmitter } from '@stencil/core';

@Component({
  tag: 'b2b-breadcrumb-item',
  styleUrl: 'breadcrumb.scss',
  shadow: true,
})
export class BreadCrumbItemComponent {
  /** If defined, an anchor tag will be rendered instead of a span, opening the specified link in the same context when clicked. */
  @Prop() href: string = null;

  /** Emits whenever a breadcrumb item is clicked and no href is specified */
  @Event({ eventName: 'b2b-selected' })
  b2bChange: EventEmitter<void>;

  private onClick = () => {
    this.b2bChange.emit();
  };

  private onKeyDown = (event: KeyboardEvent) => {
    if (event.key != 'enter') {
      return;
    }
    this.b2bChange.emit();
  };

  render() {
    return (
      <Host
        onClick={this.onClick}
        onKeyDown={this.onKeyDown}
        tabindex={Boolean(this.href) ? null : 0}>
        <span class="b2b-breadcrumb-nav__item">
          {this.href != null ? (
            <a href={this.href}>
              <slot></slot>
            </a>
          ) : (
            <slot></slot>
          )}
        </span>
      </Host>
    );
  }
}
