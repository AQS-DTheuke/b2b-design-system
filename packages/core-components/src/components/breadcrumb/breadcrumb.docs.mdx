import {
  Story,
  ArgsTable,
  PRIMARY_STORY,
  Primary,
  Canvas,
} from '@storybook/addon-docs';

# Breadcrumb Component

The Breadcrumb Component is used to display a breadcrumb trail for the user to navigate through the application. It is used in conjunction with the Breadcrumb Item Component.

## Default Arguments

<Canvas columns={3} withSource="open" withToolbar={false}>
  <Story id="components-interaction-breadcrumb--story-010-default" />
</Canvas>

The b2b-breadcrumb-item has an optional href attribute that can be used to make the breadcrumb item a link. If the href attribute is not specified, the breadcrumb item will be a span.
