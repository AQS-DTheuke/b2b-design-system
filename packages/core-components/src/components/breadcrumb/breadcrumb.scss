@use 'sass:color';
@use '../../global/b2b-styles';



:host {
  .obc_breadcrumb-nav {
    position: sticky;
    top: 3.75rem;
    z-index: 99;
    display: flex;
    flex-flow: row nowrap;
    padding: .75rem 0;
    margin-top: -24px;
    margin-bottom: 1.125rem;
    background-color: #f6f6f6;
    border-bottom: 1px solid #c4c4c4
  }
}

.obc_breadcrumb-nav__item {
  font-size: 16px;
  line-height: 24px;
  font-size: 1rem;
  line-height: 1.5rem;
  line-height: 2.25em;
  display: inline-flex;
  color: #777
}

.obc_breadcrumb-nav__item a {
  cursor: pointer;
  color: #777;
  text-decoration: none
}

.obc_breadcrumb-nav__item a:hover {
  color: #d4021d
}

.obc_breadcrumb-nav__item-after {
  color: #777;
  margin: 0 calc(0.75rem / 2);
}

.obc_breadcrumb-nav__item:last-child:after {
  display: none
}