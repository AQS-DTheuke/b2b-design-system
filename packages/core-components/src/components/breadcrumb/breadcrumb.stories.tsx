import { Meta, Story } from '@storybook/web-components';
import { html } from 'lit-html';
import breadcrumbDocs from './breadcrumb.docs.mdx';
import { getArgTypes } from '../../docs/config/utils';

const Template: Story = ({}) => {
  return html` <b2b-breadcrumb>
    <b2b-breadcrumb-item href="https://www.otto.de">Start</b2b-breadcrumb-item>
    <b2b-breadcrumb-item>Weiter</b2b-breadcrumb-item>
    <b2b-breadcrumb-item>Ende</b2b-breadcrumb-item>
  </b2b-breadcrumb>`;
};

export const story010Default = Template.bind({});
story010Default.storyName = 'Default Breadcrumb';

const anchorArgs = getArgTypes('b2b-breadcrumb');

export default {
  title: 'Components/Interaction/Breadcrumb',
  argTypes: anchorArgs,
  viewMode: 'docs',
  parameters: {
    docs: {
      page: breadcrumbDocs,
    },
  },
} as Meta;
