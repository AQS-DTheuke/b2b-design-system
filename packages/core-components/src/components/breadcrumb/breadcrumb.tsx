import { Component, Host, h } from '@stencil/core';

@Component({
  tag: 'b2b-breadcrumb',
  styleUrl: 'breadcrumb.scss',
  shadow: true,
})
export class BreadCrumbComponent {
  render() {
    return (
      <Host>
        <div class="b2b-breadcrumb-nav">
          <slot></slot>
        </div>
      </Host>
    );
  }
}
