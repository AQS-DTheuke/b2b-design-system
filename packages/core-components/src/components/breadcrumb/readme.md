# b2b-anchor



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                                                                                                | Type     | Default |
| -------- | --------- | -------------------------------------------------------------------------------------------------------------------------- | -------- | ------- |
| `href`   | `href`    | If defined, an anchor tag will be rendered instead of a span, opening the specified link in the same context when clicked. | `string` | `null`  |


## Events

| Event          | Description                                                          | Type                |
| -------------- | -------------------------------------------------------------------- | ------------------- |
| `b2b-selected` | Emits whenever a breadcrumb item is clicked and no href is specified | `CustomEvent<void>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
