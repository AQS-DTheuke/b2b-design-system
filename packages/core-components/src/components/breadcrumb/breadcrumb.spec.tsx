import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { BreadCrumbComponent } from './breadcrumb';
import { BreadCrumbItemComponent } from './breadcrumb-item';

it('should render the breadcrumb component with default values', async () => {
  const page = await newSpecPage({
    components: [BreadCrumbComponent, BreadCrumbItemComponent],
    template: () => (
      <b2b-breadcrumb>
        <b2b-breadcrumb-item href="https://www.otto.de">
          First
        </b2b-breadcrumb-item>
        <b2b-breadcrumb-item>Second</b2b-breadcrumb-item>
      </b2b-breadcrumb>
    ),
  });

  expect(page.root).toMatchSnapshot();
});
