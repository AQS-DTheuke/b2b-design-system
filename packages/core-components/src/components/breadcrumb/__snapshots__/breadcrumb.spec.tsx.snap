// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the breadcrumb component with default values 1`] = `
<b2b-breadcrumb>
  <mock:shadow-root>
    <div class="b2b-breadcrumb-nav">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <b2b-breadcrumb-item>
    <mock:shadow-root>
      <span class="b2b-breadcrumb-nav__item">
        <a href="https://www.otto.de">
          <slot></slot>
        </a>
      </span>
    </mock:shadow-root>
    First
  </b2b-breadcrumb-item>
  <b2b-breadcrumb-item tabindex="0">
    <mock:shadow-root>
      <span class="b2b-breadcrumb-nav__item">
        <slot></slot>
      </span>
    </mock:shadow-root>
    Second
  </b2b-breadcrumb-item>
</b2b-breadcrumb>
`;
